### DISCUSSION TITLE
Wrong pytorch

### DISCUSSION THREAD

**Discussion by Creepybits** [facebookresearch/xformers#1002] [Q&A]
I'm sorry if this question isn't supposed to go here, but I'll give it a go.

I had PyTorch 2.1.2+cu121 installed and got a warning saying that

`xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.2.1+cu121 with CUDA 1201 (you have 2.1.2+cu121)`

So I uninstalled it and did a clean install of PyTorch 2.2.1+cu121, and instead get the message:
`xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.1.2+cu121 with CUDA 1201 (you have 2.2.1+cu121)`

That doesn't seem fair at all! 

And on top of it all I get an error message saying that "procedure" something issues with

`?requires_grad@TensorOptions@c10@@QEBA?AU12@V?$optional@_N@2@@Z`

at 

\AppData\Local\Programs\Python310\Lib\site-packagees\xformers\_C.pyd

I have no idea what any of this means, and I cant seem to fix it no matter what, I'm sure it's probably something super easy, but I don't have any knowledge at all in this area

**Comment by danthe3rd** [comment]
Hi,
How did you install Xformers and PyTorch originally?
I would try to uninstall Xformers and reinstall it with:
```
pip uninstall -y xformers torch
pip install -U xformers torch --index-url https://download.pytorch.org/whl/cu121
```

**Reply by Creepybits** [reply]
Im not 100% sure, but most likely they got installed together with Automatic1111

I tried your suggestion, it didn't make any difference

**Comment by Creepybits** [comment]
My xformer problems got solved while I was fixing another thing:  [stable-cascade-one-click-installer](https://github.com/EtienneDosSantos/stable-cascade-one-click-installer/discussions/33#discussioncomment-8863612)
